enterprise/kernel/src/main/java/org/neo4j/kernel/impl/pagecache/Profile.java

```diff
@@ -0,0 +1,170 @@
+import java.util.Objects;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+import java.util.zip.GZIPInputStream;
+import java.util.zip.GZIPOutputStream;
+
+import org.neo4j.io.IOUtils;
+import org.neo4j.io.fs.FileSystemAbstraction;
+import org.neo4j.io.pagecache.PagedFile;
+
+import static org.neo4j.kernel.impl.pagecache.PageCacheWarmer.SUFFIX_CACHEPROF;
+
+final class Profile implements Comparable<Profile>
+{
+    private final File profileFile;
+    private final File pagedFile;
+    private final long profileCount;
+
+    private Profile( File profileFile, File pagedFile, long profileCount )
+    {
+        Objects.requireNonNull( profileFile );
+        Objects.requireNonNull( pagedFile );
+        this.profileFile = profileFile;
+        this.pagedFile = pagedFile;
+        this.profileCount = profileCount;
+    }
+
+    @Override
+    public int compareTo( Profile that )
+    {
+        int compare = profileFile.compareTo( that.profileFile );
+        return compare == 0 ? Long.compare( profileCount, that.profileCount ) : compare;
+    }
+
+    @Override
+    public boolean equals( Object o )
+    {
+        if ( o instanceof Profile )
+        {
+            Profile profile = (Profile) o;
+            return profileFile.equals( profile.profileFile );
+        }
+        return false;
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return profileFile.hashCode();
+    }
+
+    File file()
+    {
+        return profileFile;
+    }
+
+    void delete( FileSystemAbstraction fs )
+    {
+        fs.deleteFile( profileFile );
+    }
+
+    InputStream read( FileSystemAbstraction fs ) throws IOException
+    {
+        InputStream source = fs.openAsInputStream( profileFile );
+        try
+        {
+            return new GZIPInputStream( source );
+        }
+        catch ( IOException e )
+        {
+            IOUtils.closeAllSilently( source );
+            throw new IOException( "Exception when building decompressor.", e );
+        }
+    }
+
+    OutputStream write( FileSystemAbstraction fs ) throws IOException
+    {
+        OutputStream sink = fs.openAsOutputStream( profileFile, false );
+        try
+        {
+            return new GZIPOutputStream( sink );
+        }
+        catch ( IOException e )
+        {
+            IOUtils.closeAllSilently( sink );
+            throw new IOException( "Exception when building compressor.", e );
+        }
+    }
+
+    Profile next()
+    {
+        long nextCount = profileCount + 1L;
+        return new Profile( profileName( pagedFile, nextCount ), pagedFile, nextCount );
+    }
+
+    static Profile first( File file )
+    {
+        return new Profile( profileName( file, 0 ), file, 0 );
+    }
+
+    private static File profileName( File file, long count )
+    {
+        String name = file.getName();
+        File dir = file.getParentFile();
+        return new File( dir, "." + name + "." + Long.toHexString( count ) + SUFFIX_CACHEPROF );
```

Comment:
Wouldn't it be simpler to just use base 10?

Reply:
More compact names.